File: LeetCode/Easy/1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.cpp
class Solution {
public:
    string largestOddNumber(string num) {
        
        int index = -1;
        for(int j=num.length()-1 ; j>=0 ; j--){

            if((num[j]-'0') % 2 == 1){
                index = j;
                break;
            }
        }   
        return num.substr(0,index+1);
    }
};

File: LeetCode/Easy/1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {

        int endIdx = -1;
        for(int i=num.length()-1 ; i>=0; i--){

            if((num.charAt(i)-'0')%2==1){
                endIdx = i;
                break;
            }
        }

        // // remove leading zeros
        // int startIdx = 0;
        // while(startIdx < num.length()){
        //     if(num.charAt(startIdx)=='0') startIdx++;
        //     else break;
        // }

        return num.substring(0, endIdx+1);
    
    }
}